src/ec/packages/ec-component-button/button.story.js

```diff
@@ -28,15 +28,53 @@ uiIcons.forEach(icon => {
   iconsList[icon] = icon;
 });
 
-const prepareButton = data => {
-  const dataWithIcon = data;
-  dataWithIcon.icon = {
-    type: 'ui',
-    path: defaultSprite,
-    size: 'xs',
+// Labels for the groups.
+const requiredGroupId = 'Mandatory elements';
+const optionalGroupId = 'Optional elements';
+
+// Preserve the adapted specs.
```

Comment:
@Joosthe , why do you need this "cloning" of the input data? This is something we need only when we create buttons to hide/show something, because we need to retrieve back the original value from the specs, but here i don't see why we would need it. Apparently you are applying the complexity of other use cases where we needed to find solutions to specific issues, also to the standard use cases like this. You don't need to apply logic to this knobs, there is no such use case like the "label" will not be there, so why defining the knob like this? To me you should reuse everything from the data you have and only override what you need, because you want to set a knob, for instance, by redefining all the properties in the object instead you then need to set again things that are already in place (icon, for instance).
Doing like this, you would receive the data and you would then return the same data after applying the changes you want, like in the adapter, here you don't know anymore what was in the data that came from the specs, instead.